sparse-traits/src/spaces/linear_space.rs

```diff
@@ -13,15 +13,19 @@ pub trait LinearSpace {
     type F: Scalar;
 
     /// Type associated with elements of the space.
-    type E: Element<Space = Self>;
+    type E<'b>: Element<Space = Self> where Self: 'b;
 
     /// Create a new vector from the space.
-    fn create_element(&self) -> Self::E {
+    fn create_element<'b>(&'b self) -> Self::E<'b> {
         std::unimplemented!();
     }
 
     /// Norm of a vector.
-    fn norm(_x: ElementView<Self>, _res: &mut <Self::F as Scalar>::Real) -> Result<()> {
+    fn norm<'b>(
+        &'b self,
+        _x: ElementView<'b, 'b, Self>,
```

Comment:
Two lifetimes are not necessary in this context and I have fixed ElementView to only use one lifetime. Basically, the two lifetimes are that of the element and that of the view on the element. But since the lifetime of the view is always shorter than the lifetime on the element and we only need the view, we can just have a single lifetime parameter signifying the lifetime of the view.